### ISSUE TITLE
`quotes` with `allowEscape: true` doesn't catch `'"'"'`

### ISSUE THREAD

**Issue by Alhadis** [eslint/eslint#10726]
**Tell us about your environment**

* ESLint version: `v5.2.0`
* Node version: `v8.9.4`
* npm version: `v5.6.0`

**What parser (default, Babel-ESLint, etc.) are you using?**
Default.

**Please show your full configuration:**
<details>
<summary>Configuration</summary>
<b>NB:</b> Also available at <a href="https://github.com/Alhadis/.files/blob/master/.eslintrc.json"><code>Alhadis/.files/.eslintrc.json</code></a>.

~~~json
{
	"extends": "eslint:recommended",
	"env": {
		"browser": true,
		"es6": true,
		"mocha": true,
		"node": true
	},
	"parserOptions": {
		"ecmaVersion": 2017,
		"sourceType": "module"
	},
	"rules": {
		"eqeqeq": [
			"error",
			"smart"
		],
		"eol-last": [
			"error",
			"always"
		],
		"indent": [
			"error",
			"tab",
			{
				"ignoreComments": true,
				"VariableDeclarator": 0,
				"SwitchCase": 1
			}
		],
		"linebreak-style": [
			"error",
			"unix"
		],
		"no-fallthrough": [
			"error",
			{"commentPattern": "[Ff]alls?[-\\s]?through"}
		],
		"no-return-await": "error",
		"no-case-declarations": 0,
		"no-cond-assign": 0,
		"no-console": 0,
		"no-empty": 0,
		"quotes": [
			"error",
			"double",
			{"avoidEscape": true}
		],
		"semi": [
			"error",
			"always"
		],
		"spaced-comment": [
			"error",
			"always"
		],
		"yoda": [
			"error",
			"always",
			{"onlyEquality": true}
		]
	}
}
~~~
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

~~~js
/**
 * Locate a program file in the user's $PATH.
 *
 * Resolves with an empty string/array if nothing was found.
 *
 * @example which("curl") == "/usr/bin/curl"
 * @example which("nada") == ""
 * @param {String} name
 * @param {Boolean} [all=false]
 * @return {Promise}
 */
async function which(name, all = false){
	if(!name) return all ? [] : "";
	const {exec} = require("child_process");
	const cmdStr = "win32" === process.platform
		? `@for %g in (ECHO ${name.replace(/%/g, "%%")}) do`
			+ " @for %e in (%PATHEXT%) do"
			+ " @for %i in (%g%e) do "
			+ ' @if NOT "%~$PATH:i"=="" echo %~$PATH:i'
		: all
			? "IFS=:; for i in $PATH; do"
				+ ` p="$i/"'${name.replace(/'/g, `'"'"'`)}';`
				+ ' if [ -x "$p" ] && [ -f "$p" ]; then printf "%s\\n" "$p"; fi;'
				+ " done"
			: `command -v '${name.replace(/'/g, `'"'"'`)}' 2>/dev/null || true`;
	const result = await new Promise((resolve, reject) =>
		exec(cmdStr, {windowsHide: true}, (error, output) => error
			? reject(error)
			: resolve(output.split(/\r?\n/).filter(Boolean))));
	return all ? result : result[0] || "";
}
~~~

This is just [a helper function](https://github.com/Alhadis/Utils/blob/4ee816be3a8ccd84b5d0dbd5f6000a123c8b8dbe/lib/node.js#L204-L234) I wrote to check if a program exists in the user's `$PATH` or not. The <code>&#96;'"'"'&#96;</code> is what's escaping each single-quote so it doesn't screw with the surrounding shell code.

**What did you expect to happen?**
I expected that <code>&#96;'"'"'&#96;</code> would be seen as a legitimate case of not wanting to use backslashes....Since both single *and* double quotes are in the same string, it logically stands to reason that one would use a template literal to avoid having to backslash either.

Note that I still want `allowTemplateLiterals` disabled, because I want to catch useless uses of templates (where they're basically just doing something an ordinary string literal should be doing).

**What actually happened? Please include the actual, raw output from ESLint.**

~~~console
λ ~/Labs/Roff.js: eslint .
/home/Alhadis/Labs/Roff.js/lib/postproc/canvas/renderer.js
  635:4  error  Expected a 'break' statement before 'case'  no-fallthrough

/home/Alhadis/Labs/Roff.js/lib/system/shell-cache.js
  277:40  error  Strings must use doublequote  quotes

✖ 2 problems (2 errors, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
~~~

I hope this is all the info you need. If not, give me a holler. =)

**Comment by jpickwell** [comment]
I'd say the real issue is that there needs to be a rule that checks for template literals that don't use interpolation and aren't being used as a way to avoid string escapes.

**Comment by Alhadis** [comment]
That would be even better, actually.

**Comment by kaicataldo** [comment]
Some prior relevant discussion: https://github.com/eslint/eslint/issues/5234.

I agree that it feels like an overlapping of responsibilities for the `quotes` rule to care about whether or not the template literal backticks _should_ be used, and I do think the issue described here should be one that the `quotes` rule should handle correctly when the configuration is set to `{ "avoidEscape": true, "allowTemplateLiterals": true }`.

It sounds like we have two proposals here - can we split the out? One is to update `quotes` so that it allows
```
const foo = `"hello" 'world'`;
```
I support this, and could even see the argument that the current behavior is a bug.

A proposal to enforce whether or not backticks should be used would need to be fleshed out more and I think should be split out into a separate issue if the author would still like to advocate for it.
